Define Mailer and port example to an exe that can crudely register stuff

ch3pjw opened the pull request:
This does quite do All The Things yet, but it can send email in reaction to subscription events _generated in the terminal_. You need to set the following environment vars to make it go: `SMTP_SERVER`, `SMTP_USERNAME`, `SMTP_PASSWORD`. If in addition you set `SMTP_LOGGING` to any non-empty value, you'll get a log from `HaskellNet-SSL` letting you know what's going on with sending the emails.

ch3pjw committed:
Define Mailer and port example to an exe that can crudely register stuff

ch3pjw committed:
Merge branch 'master' into mailer

A few niggles needed to be ironed out in that merge!

Conflicts:
	src/Registration.hs

foolswood commented on app/Main.hs:
```diff
@@ -1,8 +1,43 @@
 
 main :: IO ()
-main = getEnv "PORT" >>= someFunc . read
+-- main = getEnv "PORT" >>= someFunc . read
```
Are you intending to return to this?

ch3pjw replied:
Yes - that's the `Main` that was serving HTTP. I need to make one that goes back to doing that again. I really just wanted a temporary standalone exe that I could use to send real email in response to "requests".

foolswood replied:
Gotcha.

foolswood commented on src/Mailer.hs:
```diff
@@ -0,0 +1,110 @@
+  }
+
+instance Show MailerSettings where
+  show ms = "poop"
```
Remove or say something about what you're showing?

ch3pjw replied:
Oops. Yeah, I forgot about that. I declared it because I couldn't automatically defined one once I'd added the formatters. Do you think I should omit the password from the `Show` instance, so that one can never accidentally reveal it?

foolswood replied:
Sounds like a good plan.

foolswood reviewed (commented):
Looks good.

ch3pjw committed:
Define Show instance for MailerSettings

Not sure how to handle the link formatters. Could try to represent them as
format strings, but then lose a bit of type safety, as could define a program
where you have the wrong number of placeholders...
